src/hooks/useApproveConfirmTransaction.ts

```diff
@@ -106,25 +106,29 @@ const useApproveConfirmTransaction = ({
     isConfirming: state.confirmState === 'loading',
     isConfirmed: state.confirmState === 'success',
     handleApprove: async () => {
-      const tx = await onApprove()
-      dispatch({ type: 'approve_sending' })
-      const receipt = await tx.wait()
-      if (receipt.status) {
-        dispatch({ type: 'approve_receipt' })
-        onApproveSuccess(state)
-      } else {
+      try {
+        const tx = await onApprove()
+        dispatch({ type: 'approve_sending' })
+        const receipt = await tx.wait()
+        if (receipt.status) {
+          dispatch({ type: 'approve_receipt' })
+          onApproveSuccess(state)
+        }
```

Comment:
Are you sure this doesn't need `dispatch({ type: 'approve_error' })` and toast in case `receipt.status` is `undefined`? I agree that logically if there is no status then something wrong happened and it will throw, but not sure if there is some other edge case

Reply:
Picking up from our DMs - the previous if / else condition wasn't working as intended. The 'else' was never hit. So `dispatch({ type: 'approve_error' }) ` was never firing.

Wrapping it in a try/catch does catch & successfully trigger errors states.